app.js

```diff
@@ -0,0 +1,780 @@
+    setBossHealth(players["boss"].health - 15);
+    displayDamage(15);
+    return 15;
+  }
+  if(randomNum == 2) {
+    setBossHealth(players["boss"].health - 5);
+    displayDamage(5);
+    return 5;
+  }
+  else {
+    setBossHealth(players["boss"].health - 10);
+    displayDamage(10);
+    return 10;
+  }
+}
+
+function grow() {
+  setEnergy(players["player"].energy - 2);
+  if((players["player"].health + (3*players["player"].cardsPlayed)) < 50) {
+    setPlayerHealth(players["player"].health + (3*players["player"].cardsPlayed));
+  } else {
+    setPlayerHealth(50);
+  }
+}
+function spirit() {
+  setEnergy(players["player"].energy+1);
+}
+
+function cleanse() {
+  players["player"].status = [];
+  setProperty("player_status","text", "");
+}
+
+function club() {
+  for(var i=0; i<hand.length; i++) {
+    if(hand[i].name != "none") {
+      return;
+    }
+  }
+  setBossHealth(players["boss"].health - 10);
+  displayDamage(10);
+  return 10;
+}
+
+function fortify() {
+  if(players["player"].shield >= 5) {
+    setPlayerShield(players["player"].shield + 5);
+  }
+}
+
+//---------------------------------------------------------------------------------
+
+function bossTurn() {
+  isPlayerTurn = false;
+  if(checkLifePoints()) {
+    return;
+  }
+  setBossShield(0);
+  //pick and execute an attack
+  var bossEvent = bossAttacks[randomNumber(0,bossAttacks.length-1)];
+  setTimeout(function() {
+    setProperty("boss_description","text","Time Keeper is using " + bossEvent.name);
+  }, 1000);
+  setTimeout(function() {
+    window[bossEvent.name]();
+  },3000);
+  setTimeout(function() {
+    setProperty("boss_description","text", "");
+  }, 4000);
+  setTimeout(function() {
+    useStatus();
+    startTurn();
+  }, 5000);
+  setTimeout(function() {
+    
+  }, 5000);
+}
+
+//Boss cards
+function strike() {
+  setPlayerHealth(players["player"].health - 15);
+  displayDamage(15);
+}
+function slash() {
+  players["player"].status.push(JSON.parse(JSON.stringify(statusEffects.fire)));
+  setPlayerHealth(players["player"].health-10);
+  showStatus();
+  displayDamage(10);
+}
+
+function reinforce() {
+  setBossShield(players["boss"].shield + 10);
+  setPlayerHealth(players["player"].health - 10);
+  displayDamage(10);
+}
+
+//-----------------------------------------------------------------------------
+
+
+
+//Deck creation
+
+//data split for each "screen"
+var showCase = [
+  [cards.spirit,cards.club,cards.fortify,cards.cleanse,cards.blankCard], 
+  [cards.smite, cards.block, cards.draw,cards.heal,cards.flame], 
+  [cards.smear,cards.dash,cards.chain,cards.fortune,cards.grow]
+];
+
+//starting idx
+var showCaseIdx = 1;
+
+
+function switchShowCase(button) {
+  if (button == "right") {
+    if ((showCaseIdx + 1) > 2) {
+      showCaseIdx = 0;
+    }
+    else {
+      showCaseIdx++;
+    }
+  }
+  if (button == "left") {
+    if((showCaseIdx - 1) < 0) {
+      showCaseIdx = 2;
+    }
+    else {
+      showCaseIdx--;
+    }
+  }
```

Comment:
I think it's clearer to have a `setShowcase(num)` method instead of setting it directly here. And then if you need to do anything additional when you set the Showcase, then you can just add it in that method instead of all of these places. When you call a method that a hook defines to set some state in React, all kinds of stuff needs to happen to render the effected components, etc.

Now that this is `advanceShowCase` I think it can be clearer

```
function advanceShowCase() {
  var newShowCaseIdx = showCaseIdx + 1;

  if ( newShowCaseIdx == showCase.size) {
    setShowCase(0);
  } else { 
    setShowCase(newShowCaseIdx);
  }
  
  renderShowCase();
}
```

`renderShowCase` quickly describes what the code is doing, and now it can be shared with `reverseShowCase`